modules/component/types/pickIStateType.ts

```diff
@@ -0,0 +1,8 @@
+import {PPP} from './pickValueFromComponent'
+
+/**
+ * Picks input state type from component or component props
+ * @typeparam P Either ComponentProp or ComponentNext
```

Comment:
Not sure how I feel about `ComponentProp` being used as a symbol here. `ComponentNext` is fine because that's what the whole library is about, but using `ComponentProp` here means that while I'm reading this documentation I need to go about look at a different part of it.